nixos/modules/config/shells-environment.nix

```diff
@@ -151,12 +151,6 @@ in
 
     system.build.binsh = pkgs.bashInteractive;
 
-    # Set session variables in the shell as well. This is usually
-    # unnecessary, but it allows changes to session variables to take
-    # effect without restarting the session (e.g. by opening a new
-    # terminal instead of logging out of X11).
```

Comment:
Do not we want this any more? Or is it not possible to implement it?

Reply:
I ended up restoring this code since the existing semantics of `environment.sessionVariables` works fine for this PR. The PR only needs the new `environment.profileRelativeSessionVariables` option.